test/services/knowledgeBase.service.test.js

```diff
@@ -0,0 +1,241 @@
+import mongoose from 'mongoose';
+import fs from 'fs';
+import path from 'path';
+import { expect, sinon, useDatabase } from '../config';
+import {
+  estimateReadingTime,
+  getPostInKnowledgeBaseById,
+  addPostToKnowledgeBase,
+  updatePostInKnowledgeBase,
+  deletePostFromKnowledgeBase,
+  getOnePostFromKnowledgeBase,
+  getAllPostsFromKnowledgeBase,
+} from '../../server/services/knowledgeBase.service';
+import KnowledgeBaseFactory from '../factories/knowledgeBase.factory';
+import KnowledgeBase from '../../server/models/knowledgeBase.model';
+import User from '../../server/models/user.model';
+import UserFactory from '../factories/user.factory';
+
+const threeHundredWords = fs.readFileSync(
+  path.resolve(__dirname, '../threeHundredWords.txt'),
+  'utf8',
+);
+
+useDatabase();
+
+describe('KnowledgeBase Service', () => {
+  describe('#getPostInKnowledgeBaseById', () => {
+    const id = mongoose.Types.ObjectId();
+
+    before(async () => {
+      await addPostToKnowledgeBase(
+        KnowledgeBaseFactory.build({ _id: id, author: 'John Doe', updatedBy: id }),
+      );
+    });
+
+    it('returns a valid post by Id', async () => {
+      const post = await getPostInKnowledgeBaseById(id);
+      expect(id.equals(post.id)).to.be.eql(true);
+    });
+  });
+
+  describe('#estimateReadingTime', () => {
+    context('when text is more than 150 words', () => {
```

Comment:
Can you state the exact number of words here to enable us to change the base WORDS_LENGTH (150) without editing the test. Remember, when this is reviewed by the Namo and Leonard, they might feel the minutes are not ideal and we might need to update this according to their preference.
```suggestion
    context('with 300 words', () => {
```